ffi/diplomat/src/properties_maps.rs

```diff
@@ -6,41 +6,151 @@
 pub mod ffi {
     use crate::provider::ffi::ICU4XDataProvider;
     use alloc::boxed::Box;
-    use icu_properties::{maps, Script};
+    use icu_collections::codepointtrie::TrieValue;
+    use icu_properties::{maps, PropertiesError};
 
     use crate::errors::ffi::ICU4XError;
     use diplomat_runtime::DiplomatResult;
 
+    use icu_provider::prelude::BufferProvider;
+    use icu_provider::serde::{AsDeserializingBufferProvider, DeserializingBufferProvider};
+
     #[diplomat::opaque]
-    /// An ICU4X Unicode Set Property object, capable of querying whether a code point is contained in a set based on a Unicode property. For properties whose values fit into 16 bits.
+    /// An ICU4X Unicode Set Property object, capable of querying whether a code point is contained in a set based on a Unicode property.
+    ///
+    /// For properties whose values fit into 8 bits.
     #[diplomat::rust_link(icu::properties, Mod)]
-    pub struct ICU4XCodePointMapData16(maps::CodePointMapData<Script>);
+    #[diplomat::rust_link(icu::properties::maps::CodePointMapData, Struct)]
+    #[diplomat::rust_link(icu::properties::maps::CodePointMapDataBorrowed, Struct)]
+    pub struct ICU4XCodePointMapData8(maps::CodePointMapData<u8>);
+
+    impl ICU4XCodePointMapData8 {
+        /// Gets a map for Unicode property General_Category from a [`ICU4XDataProvider`].
+        #[diplomat::rust_link(icu::properties::maps::load_general_category, Fn)]
+        pub fn try_get_general_category(
+            provider: &ICU4XDataProvider,
+        ) -> DiplomatResult<Box<ICU4XCodePointMapData8>, ICU4XError> {
+            Self::try_get_prop_inner(provider, |p| maps::load_general_category(p))
+        }
+
+        /// Gets a map for Unicode property Bidi_Class from a [`ICU4XDataProvider`].
+        #[diplomat::rust_link(icu::properties::maps::load_bidi_class, Fn)]
+        pub fn try_get_bidi_class(
+            provider: &ICU4XDataProvider,
+        ) -> DiplomatResult<Box<ICU4XCodePointMapData8>, ICU4XError> {
+            Self::try_get_prop_inner(provider, |p| maps::load_bidi_class(p))
+        }
+
+        /// Gets a map for Unicode property East_Asian_Width from a [`ICU4XDataProvider`].
```

Comment:
Discussion: We agreed last week to put East_Asian_Width and the four other segmentation properties behind the "datagen" feature. Do we still want to do that? How would that relate to the FFI?

Reply:
Perhaps we just comment it out for now? We don't have cfgs over diplomat (yet)

Reply:
Well. I think CFGs work, but diplomat will unconditionally generate code for them

Reply:
The way we handle this elsewhere in the codebase is to have these functions panic conditionally

Reply:
I don't have strong opinions though.

Reply:
I'm okay commenting them out for now.  

If we don't comment it out, then there's a chance that we generate the FFI APIs whether or not the feature was turned or not, and that could expose APIs to users that don't call into anything and panics.  If we do comment out these APIs, we avoid that problem, at the cost of not having experimental code available over FFI (until such time Diplomat can support cfgs).  It's experimental anyways, so that seems like an acceptable constrained cost.

Reply:
I'm neutral on whether or not to include the five segmenter properties.

We should most likely remove (or comment) the CCC property, since we want to expose that via the normalizer instead.

We _may_ want to remove (or comment) the Script property once we have Script_Extensions up and running, but I'm a bit more unsure about that one since there are use cases for Script alone.

Reply:
Is there a reason we don't want the properties crate to just expose all properties?